integration-tests/src/test/java/org/apache/druid/tests/indexer/AbstractStreamIndexingTest.java

```diff
@@ -67,92 +92,147 @@
+      }
     }
-    finally {
-      doMethodTeardown(generatedTestConfig, streamEventWriter);
+    if (!map.containsKey(SERIALIZER_SPEC_DIR)) {
+      throw new IAE("Failed to find serializer spec under [%s]. Found resources are %s", resourceRoot, map);
     }
+    if (map.size() == 1) {
+      throw new IAE("Failed to find input format or parser spec under [%s]. Found resources are %s", resourceRoot, map);
+    }
+    return map;
+  }
+
+  private Closeable createResourceCloser(GeneratedTestConfig generatedTestConfig)
+  {
+    return Closer.create().register(() -> doMethodTeardown(generatedTestConfig));
   }
 
-  protected void doTestIndexDataWithInputFormatStableState() throws Exception
+  protected void doTestIndexDataStableState(
+      boolean transactionEnabled,
+      String serializerPath,
+      String parserType,
+      String specPath
+  ) throws Exception
   {
-    StreamEventWriter streamEventWriter = createStreamEventWriter(config);
-    final GeneratedTestConfig generatedTestConfig = new GeneratedTestConfig();
+    final EventSerializer serializer = jsonMapper.readValue(getResourceAsStream(serializerPath), EventSerializer.class);
+    final StreamGenerator streamGenerator = new WikipediaStreamEventStreamGenerator(
+        serializer,
+        EVENTS_PER_SECOND,
+        CYCLE_PADDING_MS
+    );
+    final GeneratedTestConfig generatedTestConfig = new GeneratedTestConfig(parserType, getResourceAsString(specPath));
     try (
-        final Closeable ignored1 = unloader(generatedTestConfig.getFullDatasourceName())
+        final Closeable closer = createResourceCloser(generatedTestConfig);
+        final StreamEventWriter streamEventWriter = createStreamEventWriter(config, transactionEnabled)
     ) {
-      final String taskSpec = generatedTestConfig.getStreamIngestionPropsTransform().apply(getResourceAsString(INDEXER_FILE_INPUT_FORMAT));
+      final String taskSpec = generatedTestConfig.getStreamIngestionPropsTransform()
+                                                 .apply(getResourceAsString(SUPERVISOR_SPEC_TEMPLATE_PATH));
       LOG.info("supervisorSpec: [%s]\n", taskSpec);
       // Start supervisor
       generatedTestConfig.setSupervisorId(indexer.submitSupervisor(taskSpec));
       LOG.info("Submitted supervisor");
       // Start data generator
-      wikipediaStreamEventGenerator.run(generatedTestConfig.getStreamName(), streamEventWriter, TOTAL_NUMBER_OF_SECOND, FIRST_EVENT_TIME);
+      streamGenerator.run(
+          generatedTestConfig.getStreamName(),
+          streamEventWriter,
+          TOTAL_NUMBER_OF_SECOND,
+          FIRST_EVENT_TIME
+      );
       verifyIngestedData(generatedTestConfig);
     }
-    finally {
-      doMethodTeardown(generatedTestConfig, streamEventWriter);
-    }
   }
 
-  void doTestIndexDataWithLosingCoordinator() throws Exception
+  void doTestIndexDataWithLosingCoordinator(boolean transactionEnabled) throws Exception
```

Comment:
Kinesis does not have transaction enabled/disabled. Does it make sense to put this at Kafka layer instead of here?

Reply:
Good point. Can I do it in a follow-up PR with the change for README.md you commented below?

Reply:
Sure